prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -290,18 +281,19 @@ class ReportFunction : Logging {
             ?: request.queryParameters[PAYLOAD_NAME_PARAMETER]
     }
 
-    // TODO: Make this so that we check sender's configuration if there is no param type, this is blocked by
-    //  adding the 'processingType' attribute to a sender
-    private fun processingType(request: HttpRequestMessage<String?>): ProcessingType {
+    private fun processingType(request: HttpRequestMessage<String?>, sender: Sender): ProcessingType {
         // uppercase first char, so it matches enum when 'async' is passed in
-        val processingTypeString = request.queryParameters.getOrDefault(PROCESSING_TYPE_PARAMETER, "Sync")
-            .replaceFirstChar { it.uppercase() }
-        val processingType = try {
-            ProcessingType.valueOf(processingTypeString)
-        } catch (e: IllegalArgumentException) {
-            ProcessingType.Sync
+        val processingTypeString = request.queryParameters.get(PROCESSING_TYPE_PARAMETER)
```

Comment:
If we trust our senders, then the priority order used here is correct.   This uses the URL param if it exists and is valid.   If it does not exist, use the sender setting. So the senders have the right to override our settting, and we can't control that.

I agree with this; we can always switch the priority should the need arise.